### DISCUSSION TITLE
Single node generator

### DISCUSSION THREAD

**Discussion by Nuriacor** [idaholab/moose#19924] [Q&A General] [answered]
Hi MOOSE,
I have a simple question. I would like to add a new boundary to a node using ExtraNodesetGenerator. However, in the original mesh there is no node in the coordinates I want to add my boundary. Is it possible to generate a single node with the mesh modifiers of MOOSE?
Cheers and thanks!
Nuria

**Comment by loganharbour** [comment, marked as answer]
You are trying to _add_ a new node at a given point and also assign it to a nodeset. Is that correct?

**Reply by Nuriacor** [reply]
yes!

**Reply by loganharbour** [reply]
What are you trying to achieve exactly? If you're adding a single node without connectivity, I suspect that you would be better off using a scalar variable with a scalar kernel

**Reply by Nuriacor** [reply]
I want to add that node and nodeset to an exciting mesh. Ultimately, I want to use that node as a sink.

**Reply by loganharbour** [reply]
What action do you plan on doing on the node (kernel, BC, etc)?

**Reply by Nuriacor** [reply]
PorousFlow 
https://mooseframework.inl.gov/source/dirackernels/PorousFlowSquarePulsePointSource.html

**Reply by Nuriacor** [reply]
But I want to add a nodeset where the PorousFlowSquarePulsePointSource is because I want to change the temperature of the node.

**Reply by GiudGiud** [reply]
Hello

There are a few options:

- would the node you want be added automatically if you refine that part of the mesh? If so the refineBlockGenerator or refineBoundaryGenerator could help you create that node before reassigning it to a nodeset 
- Are you ok with moving another node to the position you want the node to be at? If so then you can use move(or modify)NodeGenerator

Where is the node you want to add? Is it inside the existing mesh or is it outside?

guillaume

**Reply by Nuriacor** [reply]
Hi,
Thanks for your help! MoveNodeGenerator made the trick. Thanks a lot!
